src/util/amount.rs

```diff
@@ -0,0 +1,863 @@
+// To the extent possible under law, the author(s) have dedicated all
+// copyright and related and neighboring rights to this software to
+// the public domain worldwide. This software is distributed without
+// any warranty.
+//
+// You should have received a copy of the CC0 Public Domain Dedication
+// along with this software.
+// If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
+//
+
+//! Amounts
+//!
+//! Defines a type [Amount] that can be used to express Bitcoin amounts in
+//! different precisions and supports arithmetic and convertion to various
+//! denominations.
+//!
+//!
+//! Warning!
+//!
+//! In a few functions, this module supports convertion to and from floating-point numbers.
+//! Please be aware of the risks of using floating-point numbers for financial applications.
+//! These types of numbers do not give any guarantee to retain the precision
+//! of the original amount when converting, or when doing arithmetic operations.
```

Comment:
Ah yeah that's a good point. Let me quickly go over all documentation again to see if something changed.

Reply:
Updated the docs. I moved the important piece to the `Amount` type instead of the module because I think most users will find this amount through a re-export as `bitcoin::Amount`.

Reply:
updated doc looks good to me